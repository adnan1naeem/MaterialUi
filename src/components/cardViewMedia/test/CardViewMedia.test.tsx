/* import { CardViewMedia } from '../'
import renderer from 'react-test-renderer';
import React from 'react';
import "jest";
import { BlueRain, BlueRainConsumer } from '@blueeast/bluerain-os';
function handleTextInputChange() {
    console.log('I am clicked');
}
test('Cardview componenet overlay title direction top left', () => {
    const cardviewmedia = renderer.create(
        <BlueRainConsumer>
            {(BR: BlueRain) =>
                <CardViewMedia
                    overlayTitle={<BR.Components.View ><BR.Components.Overline style={{ color: 'white' }}>Overline</BR.Components.Overline></BR.Components.View>}
                    overlaySubTiltle={<BR.Components.View ><BR.Components.H5 style={{ color: 'white' }}>Basics of Brand Identity</BR.Components.H5></BR.Components.View >}
                    overlayOverline={<BR.Components.View ><BR.Components.Subtitle2 style={{ color: 'white' }}>Subtitle2</BR.Components.Subtitle2></BR.Components.View >}
                    overlayColor={'rgba(30,24,40,0.35)'}
                    mediaComponent='img'
                    overlayDirection='topLeft'
                    mediaWidth={250}
                    mediaHeight={250}
                    src="https://images.unsplash.com/photo-1541828985935-1fe979f9fc0b?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=42ae3092ed745c9b579463975f7494ad&auto=format&fit=crop&w=500&q=60"
                />

            }
        </BlueRainConsumer >
    ).toJSON();
    expect(cardviewmedia.props.overlayDirection).toEqual('topeft');
    expect(cardviewmedia).toMatchSnapshot();
});
test('Cardview componenet overlay title with top right direction', () => {
    const cardviewmedia = renderer.create(
        <BlueRainConsumer>
            {(BR: BlueRain) =>
                <CardViewMedia

                    overlayTitle={<BR.Components.View ><BR.Components.Overline style={{ color: 'white' }}>Overline</BR.Components.Overline></BR.Components.View>}
                    overlaySubTiltle={<BR.Components.View ><BR.Components.H5 style={{ color: 'white' }}>Basics of Brand Identity</BR.Components.H5></BR.Components.View >}
                    overlayOverline={<BR.Components.View ><BR.Components.Subtitle2 style={{ color: 'white' }}>Subtitle2</BR.Components.Subtitle2></BR.Components.View >}
                    overlayColor={'rgba(30,24,40,0.35)'}
                    mediaComponent='img'
                    overlayDirection='topRight'
                    mediaWidth={250}
                    mediaHeight={250}
                    src="https://images.unsplash.com/photo-1541828985935-1fe979f9fc0b?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=42ae3092ed745c9b579463975f7494ad&auto=format&fit=crop&w=500&q=60"
                />

            }
        </BlueRainConsumer >
    ).toJSON();
    expect(cardviewmedia.props.overlayDirection).toEqual('topRIght');
    expect(cardviewmedia).toMatchSnapshot();

});
test('Cardviewmeida component image and  overlay title in bottom left  ', () => {
    const cardviewmedia = renderer.create(
        <BlueRainConsumer>
            {(BR: BlueRain) =>
                <CardViewMedia

                    overlayTitle={<BR.Components.View ><BR.Components.Overline style={{ color: 'white' }}>Overline</BR.Components.Overline></BR.Components.View>}
                    overlaySubTiltle={<BR.Components.View ><BR.Components.H5 style={{ color: 'white' }}>Basics of Brand Identity</BR.Components.H5></BR.Components.View >}
                    overlayOverline={<BR.Components.View ><BR.Components.Subtitle2 style={{ color: 'white' }}>Subtitle2</BR.Components.Subtitle2></BR.Components.View >}
                    overlayColor={'rgba(30,24,40,0.35)'}
                    mediaComponent='img'
                    overlayDirection='bottomLeft'
                    mediaWidth={250}
                    mediaHeight={250}
                    src="https://images.unsplash.com/photo-1541828985935-1fe979f9fc0b?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=42ae3092ed745c9b579463975f7494ad&auto=format&fit=crop&w=500&q=60"
                />

            }
        </BlueRainConsumer >
    ).toJSON();
    expect(cardviewmedia.props.overlayDirection).toEqual('bottomLeft');
    expect(cardviewmedia).toMatchSnapshot();
});
test('Cardviewmeida component image and  overlay title in bottom right  ', () => {
    const cardviewmedia = renderer.create(
        <BlueRainConsumer>
            {(BR: BlueRain) =>
                <CardViewMedia

                    overlayTitle={<BR.Components.View ><BR.Components.Overline style={{ color: 'white' }}>Overline</BR.Components.Overline></BR.Components.View>}
                    overlaySubTiltle={<BR.Components.View ><BR.Components.H5 style={{ color: 'white' }}>Basics of Brand Identity</BR.Components.H5></BR.Components.View >}
                    overlayOverline={<BR.Components.View ><BR.Components.Subtitle2 style={{ color: 'white' }}>Subtitle2</BR.Components.Subtitle2></BR.Components.View >}
                    overlayColor={'rgba(30,24,40,0.35)'}
                    mediaComponent='img'
                    overlayDirection='bottomRight'
                    mediaWidth={250}
                    mediaHeight={250}
                    src="https://images.unsplash.com/photo-1541828985935-1fe979f9fc0b?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=42ae3092ed745c9b579463975f7494ad&auto=format&fit=crop&w=500&q=60"
                />

            }
        </BlueRainConsumer >
    ).toJSON();
    expect(cardviewmedia.props.overlayDirection).toEqual('bottomRight');
    expect(cardviewmedia).toMatchSnapshot();
});
test('Cardviewmeida component video and  overlay title in bottom right  ', () => {
    const cardviewmedia = renderer.create(
        <BlueRainConsumer>
            {(BR: BlueRain) =>
                <CardViewMedia

                    overlayTitle={<BR.Components.View ><BR.Components.Overline style={{ color: 'white' }}>Overline</BR.Components.Overline></BR.Components.View>}
                    overlaySubTiltle={<BR.Components.View ><BR.Components.H5 style={{ color: 'white' }}>Basics of Brand Identity</BR.Components.H5></BR.Components.View >}
                    overlayOverline={<BR.Components.View ><BR.Components.Subtitle2 style={{ color: 'white' }}>Subtitle2</BR.Components.Subtitle2></BR.Components.View >}
                    overlayColor={'rgba(30,24,40,0.35)'}
                    mediaComponent='img'
                    overlayDirection='bottomRight'
                    mediaWidth={250}
                    mediaHeight={250}
                    src='http://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny.mp4'
                />

            }
        </BlueRainConsumer >
    ).toJSON();
    expect(cardviewmedia.props.overlayDirection).toEqual('bottomRight');
    expect(cardviewmedia).toMatchSnapshot();
}); */