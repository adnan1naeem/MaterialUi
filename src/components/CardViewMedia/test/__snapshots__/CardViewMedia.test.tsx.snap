// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly CardMedia with overlay title in bottom left 1`] = `
<View
  style={
    Object {
      "height": 250,
      "width": 250,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "height": 250,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 250,
      }
    }
  >
    <ExponentVideo
      nativeResizeMode="mock"
      onErrorNative={[Function]}
      onFullscreenUpdateNative={[Function]}
      onLoadNative={[Function]}
      onLoadStartNative={[Function]}
      onReadyForDisplayNative={[Function]}
      onStatusUpdateNative={[Function]}
      resizeMode="cover"
      shouldPlay={true}
      source={
        Object {
          "overridingExtension": null,
          "uri": "http://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny.mp4",
        }
      }
      status={
        Object {
          "shouldPlay": true,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "column",
        "height": 250,
        "justifyContent": "flex-end",
        "width": 250,
      }
    }
  >
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        overlaytitle
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        overlaySubTiltle
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        overlayOverline
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly CardMedia with overlay title in top bottom right 1`] = `
<View
  style={
    Object {
      "height": 250,
      "width": 250,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "height": 250,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 250,
      }
    }
  >
    <ExponentVideo
      nativeResizeMode="mock"
      onErrorNative={[Function]}
      onFullscreenUpdateNative={[Function]}
      onLoadNative={[Function]}
      onLoadStartNative={[Function]}
      onReadyForDisplayNative={[Function]}
      onStatusUpdateNative={[Function]}
      resizeMode="cover"
      shouldPlay={true}
      source={
        Object {
          "overridingExtension": null,
          "uri": "http://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny.mp4",
        }
      }
      status={
        Object {
          "shouldPlay": true,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "column",
        "height": 250,
        "justifyContent": "flex-end",
        "width": 250,
      }
    }
  >
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        overlaytitle
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        overlaySubTiltle
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        overlayOverline
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly CardMedia with overlay title in top left position 1`] = `
<View
  style={
    Object {
      "height": 250,
      "width": 250,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "height": 250,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 250,
      }
    }
  >
    <ExponentVideo
      nativeResizeMode="mock"
      onErrorNative={[Function]}
      onFullscreenUpdateNative={[Function]}
      onLoadNative={[Function]}
      onLoadStartNative={[Function]}
      onReadyForDisplayNative={[Function]}
      onStatusUpdateNative={[Function]}
      resizeMode="cover"
      shouldPlay={true}
      source={
        Object {
          "overridingExtension": null,
          "uri": "http://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny.mp4",
        }
      }
      status={
        Object {
          "shouldPlay": true,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "column",
        "height": 250,
        "justifyContent": "flex-start",
        "width": 250,
      }
    }
  >
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        overlaytitle
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        overlaySubTiltle
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        overlayOverline
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly CardMedia with overlay title in top left right 1`] = `
<View
  style={
    Object {
      "height": 250,
      "width": 250,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "height": 250,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 250,
      }
    }
  >
    <ExponentVideo
      nativeResizeMode="mock"
      onErrorNative={[Function]}
      onFullscreenUpdateNative={[Function]}
      onLoadNative={[Function]}
      onLoadStartNative={[Function]}
      onReadyForDisplayNative={[Function]}
      onStatusUpdateNative={[Function]}
      resizeMode="cover"
      shouldPlay={true}
      source={
        Object {
          "overridingExtension": null,
          "uri": "http://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny.mp4",
        }
      }
      status={
        Object {
          "shouldPlay": true,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "column",
        "height": 250,
        "justifyContent": "flex-start",
        "width": 250,
      }
    }
  >
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        overlaytitle
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        overlaySubTiltle
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        overlayOverline
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly Cardviewmedia without any prop 1`] = `
<View
  style={
    Object {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "height": undefined,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": undefined,
      }
    }
  >
    <ExponentVideo
      nativeResizeMode="mock"
      onErrorNative={[Function]}
      onFullscreenUpdateNative={[Function]}
      onLoadNative={[Function]}
      onLoadStartNative={[Function]}
      onReadyForDisplayNative={[Function]}
      onStatusUpdateNative={[Function]}
      resizeMode="cover"
      shouldPlay={true}
      source={null}
      status={
        Object {
          "shouldPlay": true,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "padding": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      />
    </View>
  </View>
</View>
`;
